src/gui/guiFormSpecMenu.cpp

```diff
@@ -270,6 +270,25 @@ v2s32 GUIFormSpecMenu::getElementBasePos(bool absolute,
 	return v2s32(pos_f.X, pos_f.Y);
 }
 
+v2s32 GUIFormSpecMenu::getRealCoordinateBasePos(bool absolute,
+		const std::vector<std::string> *v_pos)
```

Comment:
ref not pointer

Reply:
`getElementBasePos` uses a pointer, too. I guess both should be changed.

Reply:
Yes, I pretty much copied and pasted and changed the logic.  If I was to change them, how should I do it?  I'm still not very experienced in C++.

Reply:
Make the asterisk(*) here before `v_pos` to an ampersand(&) and treat the parameter as if it was a "normal" parameter, ie. remove the dereferencing asterisks in the function body and the referencing ampersands in the function calls.

Reply:
```diff
-	pos = getRealCoordinateBasePos(false, &v_pos);
+	pos = getRealCoordinateBasePos(false, v_pos);

-	const std::vector<std::string> *v_pos)
+	const std::vector<std::string> &v_pos)

-	pos_f.X += stof((*v_pos)[0]) + pos_offset.X;
-	pos_f.Y += stof((*v_pos)[1]) + pos_offset.Y;
+	pos_f.X += stof(v_pos[0]) + pos_offset.X;
+	pos_f.Y += stof(v_pos[1]) + pos_offset.Y;
```

`getElementBasePos()` takes a pointer so that it can be used without a position value (`parseSimpleField()` and `parseLabel()`).